.github/workflows/gh-workflow-approve.yaml

```diff
@@ -1,5 +1,6 @@
 ---
 name: Approve GitHub Workflows
+permissions: read-all
```

Comment:
Probably something like below?

```
permissions:
  actions:write
```

Reply:
That should work!

Reply:
I was initially going to set it with explicit permission at the top of the file. However, I based this implementation on other workflows. For example, the scorecard workflow declares the `read-all` permission at the top of the file:

https://github.com/etcd-io/etcd/blob/f00ceb6c0125e8e1f925cdc3cb02e5f472863577/.github/workflows/scorecards.yml#L11-L12

Then, in the job, sets the write permissions:

https://github.com/etcd-io/etcd/blob/f00ceb6c0125e8e1f925cdc3cb02e5f472863577/.github/workflows/scorecards.yml#L18-L22

I tested it on my fork, and seems to work fine: https://github.com/ivanvc/etcd/actions/runs/10115486985/job/27976358983?pr=210


Let me know if you still want me to change this.

Reply:
Oh, I just realised that you already added at the job level,
```
permissions:
  actions:write
```
https://github.com/ivanvc/etcd/blob/5a02298ad5a947214ba02655b0a93ac01d4c178a/.github/workflows/gh-workflow-approve.yaml#L20-L21

Looks good to me.